packages/x-charts/src/ChartsAxisHighlight/ChartsAxisHighlight.tsx

```diff
@@ -37,16 +37,34 @@ export const ChartsAxisHighlightPath = styled('path', {
   name: 'MuiChartsAxisHighlight',
   slot: 'Root',
   overridesResolver: (_, styles) => styles.root,
-})<{ ownerState: { axisHighlight: AxisHighlight } }>(({ ownerState, theme }) => ({
+})<{ ownerState: { axisHighlight: AxisHighlight } }>(({ theme }) => ({
   pointerEvents: 'none',
-  ...(ownerState.axisHighlight === 'band' && {
-    fill: theme.palette.mode === 'light' ? 'gray' : 'white',
-    fillOpacity: 0.1,
-  }),
-  ...(ownerState.axisHighlight === 'line' && {
-    strokeDasharray: '5 2',
-    stroke: theme.palette.mode === 'light' ? '#000000' : '#ffffff',
-  }),
+  variants: [
+    {
+      props: {
+        axisHighlight: 'band',
+      },
```

Comment:
This doesn't seem right, since `axisHighlight` is part of the `ownerState` prop: https://github.com/mui/mui-x/blob/8a75b228b616b1dceb24f85721340db41fc48e8a/packages/x-charts/src/ChartsAxisHighlight/ChartsAxisHighlight.tsx#L116

@siriwatknp How should we approach this?
Would this work?

```
variants: [
  {
    props: {
      ownerState: {
        axisHighlight: 'band',
      }
    },
```

Or should we do this instead?

```diff
<ChartsAxisHighlightPath
  ownerState={{ axisHighlight: 'band' }}
+ axisHighlight="band"
/>
```

Reply:
Or this?

```
variants: [
  {
    props: ({ ownerState }) => ownerState.axisHighlight === 'band',
```

Reply:
@cherniavskii, this should work:

```js
variants: [
  {
    props: { axisHighlight: 'band' }
```

The owner state values are checked in the variant's props check ([source code](https://github.com/mui/material-ui/blob/next/packages/mui-system/src/createStyled/createStyled.js#L69)). The owner state values are also available in the prop callback ([source code](https://github.com/mui/material-ui/blob/next/packages/mui-system/src/createStyled/createStyled.js#L63-L64)). This is also what we're doing in Material UI.

Let me know if it doesn't work 😊.

Reply:
AFAIK, I did thorough testing of the changes and my solution seemed to work.
It is done in multiple places on Pickers codebase.